dom.go

```diff
@@ -121,13 +121,19 @@ func (e Node) HTML() template.HTML {
 
 func (e Node) buildHTML(sb *strings.Builder) *strings.Builder {
 	if e.Name == "" {
+		// buildChildrenHTML (inline to save 32B and 1 alloc)
 		if e.InnerHTML != "" {
 			sb.WriteString(string(e.InnerHTML))
-			return sb
+		} else if e.InnerText != "" {
+			sb.WriteString(template.HTMLEscapeString(e.InnerText))
+		} else {
+			for _, child := range e.Children {
+				child.buildHTML(sb)
+			}
 		}
```

Comment:
the code inside this block is now identical to the code few lines below, marked by `// buildChildrenHTML` too